extensions/amp-story/1.0/test/test-amp-story-viewer-messaging-handler.js

```diff
@@ -119,6 +123,58 @@ describes.fakeWin('amp-story-viewer-messaging-handler', {}, env => {
     });
   });
 
+  describe('onDocumentState', () => {
+    it('should throw if no state', async () => {
+      expectAsyncConsoleError(/Invalid 'state' parameter/i, 1);
+      const subscribeStub = env.sandbox.stub(storeService, 'subscribe');
+
+      fakeViewerService.receiveMessage('onDocumentState', undefined);
+
+      expect(subscribeStub).to.not.have.been.called;
+    });
+
+    it('should throw if invalid state', async () => {
+      expectAsyncConsoleError(/Invalid 'state' parameter/i, 1);
+      const subscribeStub = env.sandbox.stub(storeService, 'subscribe');
+
+      fakeViewerService.receiveMessage('onDocumentState', {
+        state: 'UNEXISTING_STATE',
+      });
+
+      expect(subscribeStub).to.not.have.been.called;
+    });
+
+    it('should subscribe to state updates', async () => {
+      const subscribeStub = env.sandbox.stub(storeService, 'subscribe');
+
+      fakeViewerService.receiveMessage('onDocumentState', {
+        state: 'MUTED_STATE',
+      });
+
+      expect(subscribeStub).to.have.been.calledWith(StateProperty.MUTED_STATE);
+    });
+
+    it('should receive documentStateUpdate events', async () => {
+      storeService.dispatch(Action.TOGGLE_MUTED, false);
```

Comment:
is this needed?

Reply:
It is!

Reply:
oh the default is `true`, and the store won't fire the trigger if the new value is same as the old.. my bad!